server/bastionai_app/src/main.rs

```diff
@@ -96,11 +156,29 @@ impl RemoteTorch for BastionAIServer {
             .unwrap()
             .insert(identifier.clone(), module);
         let elapsed = start_time.elapsed();
-        info!("Upload Model successful in {}ms", elapsed.as_millis());
 
+        self.modules_hashes.write().unwrap().insert(
+            identifier.clone(),
+            Arc::new((model_hash.clone(), client_info.clone())),
+        );
+
+        info!(
+        target: "BastionAI",
+            
+            "Upload Model successful in {}ms", elapsed.as_millis());
+
+        telemetry::add_event(
+            TelemetryEventProps::SendModel {
+                model_name: Some(description.clone()),
+                model_hash: Some(model_hash),
+                model_size,
+                time_taken: elapsed.as_millis() as f64,
+            },
+            client_info,
+        );
         Ok(Response::new(Reference {
             identifier: format!("{}", identifier),
-            description,
+            description: description.clone(),
```

Comment:
Why is there a clone now?

Reply:
Not necessary tho.
I was cloning it up there.